crates/jstzd/src/task/endpoint.rs

```diff
@@ -0,0 +1,86 @@
+#![allow(dead_code)]
+use http::{uri::Scheme, Uri};
+
+#[derive(Debug, Clone)]
+pub struct Endpoint {
+    scheme: String,
+    host: String,
+    port: u16,
+}
+
+impl Endpoint {
```

Comment:
i think we can use this struct for OctezNode too?

Reply:
Yes, but I don't think it's that useful for now since the endpoint eventually gets transformed into a string anyway. We might update `AsyncOctezNode::config_init` so that it consumes `Endpoint` instances, but I would say we do this after other pieces of code are changed to use this async node implementation. If it's really necessary, we do it then.